### ISSUE TITLE
How to save model metadata

### ISSUE THREAD

**Issue by csukuangfj** [Tencent/ncnn#4399]
## detail | 详细描述 | 詳細な説明
Suppose I have a model like below:
```python3
import torch
import torch.nn as nn


class Foo(nn.Module):
    def __init__(
        self,
        num_layers: int,
        dim: int,
    ):
        super().__init__()
        self.dim = dim
        self.num_layers = num_layers
        layers = []
        for i in range(num_layers):
            layers.append(nn.Linear(dim, dim))

        self.layers = nn.ModuleList(layers)

    def forward(self, x: torch.Tensor):
        for layer in self.layers:
            x = layer(x)
        return x

@torch.no_grad()
def main():
    f = Foo(num_layers=2, dim=3)
    f.eval()

    x = torch.rand(1, 2, 3)
    m = torch.jit.trace(f, x)
    print(m.graph)
    m.save("m.pt")


if __name__ == "__main__":
    main()
```

```
pnnx m.pt
```

I would like to save `num_layers` and `dim` to the generated file `m.ncnn.param`, but I don't find any way to do that.

The reason is that I have models with different `num_layers` and `dim`. Once converted to ncnn, I need to get such metadata when loading them in ncnn so that I can allocate some memory to store some model internal state.

**Comment by csukuangfj** [comment]
I have several such models and don't want to provide `num_layers` and `dim`from commandline when I start the program.
Would be best to read them from `.ncnn.param` or `.ncnn.bin`.
